src/euphorie/client/browser/navigation.py

```diff
@@ -0,0 +1,105 @@
+from euphorie.client import model
+from euphorie.client.interfaces import CustomRisksModifiedEvent
+from euphorie.client.navigation import FindNextQuestion
+from euphorie.client.navigation import FindPreviousQuestion
+from plone import api
+from plone.memoize.instance import memoize
+from Products.Five import BrowserView
+from sqlalchemy import and_
+from z3c.saconfig import Session
+from zope.event import notify
+
+
+class NavigationView(BrowserView):
+    question_filter = None
+
+    @property
+    @memoize
+    def webhelpers(self):
+        return api.content.get_view("webhelpers", self.context, self.request)
+
+    @property
+    @memoize
+    def session(self):
+        return self.webhelpers.traversed_session.session
+
+    @property
+    @memoize
+    def previous_question(self):
+        return FindPreviousQuestion(
+            self.context, dbsession=self.session, filter=self.question_filter
+        )
+
+    @property
+    @memoize
+    def next_question(self):
+        return FindNextQuestion(
+            self.context, dbsession=self.session, filter=self.question_filter
+        )
+
+    def proceed_to_next(self, reply):
+        _next = reply.get("next", None)
+        # In Safari browser we get a list
+        if isinstance(_next, list):
+            _next = _next.pop()
```

Comment:
I like more what you did on the other view, where you have a dedicated `_get_next` method.
Any reason to not do the same here?

Reply:
The goal right now is only to move this from `browser/risk.py` so that we can use it for choices, too. Refactoring can be done later.